client/src/components/CreateMeetup/CreateMeetup.js

```diff
@@ -1,7 +1,246 @@
+import {
+  KeyboardDatePicker,
+  KeyboardTimePicker,
+  MuiPickersUtilsProvider,
+} from "@material-ui/pickers";
+import DayUtils from "@date-io/dayjs";
+
+const useStyles = makeStyles(() => ({
+  formSection: {
+    marginTop: 15,
+    marginBottom: 15,
+  },
+  formInput: {
+    width: 200,
+  },
+}));
 
 function CreateMeetup() {
-  return <>Create meetup</>;
+  const classes = useStyles();
+  const [name, setName] = useState("");
+  const [date, setDate] = useState(Date.now());
+  const [meetupType, setMeetupType] = useState("");
+  const [meetupLocation, setMeetupLocation] = useState("");
+  const [groupMin, setGroupMin] = useState("");
+  const [groupMax, setGroupMax] = useState("");
+  const [desc, setDesc] = useState("");
+  const [tags, setTags] = useState([]);
+
+  // TODO: load options from the server instead
+  const tagOptions = [
+    "basketball",
+    "ping pong",
+    "badminton",
+    "movie",
+    "cooking",
+  ];
+
+  const renderNameInput = () => {
+    return (
+      <div className={classes.formSection}>
+        <TextField
+          required
+          variant="outlined"
+          id="meetup-name"
+          label="Meetup Name"
+          value={name}
+          onChange={(event) => setName(event.target.value)}
+          className={classes.formInput}
+        />
+      </div>
+    );
+  };
+
+  const renderDateInput = () => {
+    return (
+      <div className={classes.formSection}>
+        <MuiPickersUtilsProvider utils={DayUtils}>
+          <KeyboardDatePicker
+            variant="inline"
+            id="date-picker-inline"
+            inputVariant="outlined"
+            format="MM/DD/YYYY"
+            label="Meetup Date"
+            value={date}
+            onChange={(newDate) => setDate(newDate)}
+            className={classes.formInput}
+            style={{
+              marginRight: 20,
+            }}
+          />
+          <KeyboardTimePicker
+            variant="inline"
+            inputVariant="outlined"
+            id="time-picker"
+            label="Meetup Time"
+            value={date}
+            onChange={(newDate) => setDate(newDate)}
+            className={classes.formInput}
+          />
+        </MuiPickersUtilsProvider>
+      </div>
+    );
+  };
+
+  const renderMeetupLocation = () => {
+    return (
+      <Box display="flex" flexWrap="wrap" className={classes.formSection}>
+        <FormControl variant="outlined" className={classes.formInput}>
+          <InputLabel id="select-meetup-type-label">Meetup Type</InputLabel>
+          <Select
+            label="Meetup Type"
+            id="select-meetup-type"
+            labelID="select-meetup-type-label"
+            value={meetupType}
+            onChange={(event) => setMeetupType(event.target.value)}
+            className={classes.formInput}
+          >
+            <MenuItem value="In person">In person</MenuItem>
+            <MenuItem value="Remote">Remote</MenuItem>
+          </Select>
```

Comment:
Having a drop down here is a nice touch.

For the `value=""`, can we make them a bit more programmer-friendly? Like "in-person" vs. "remote".

Another minor thing. `a ? b : null` in JSX could instead be written as `a && b`.

Reply:
Oh good point about the `a && b` thing!